File: Problem_Sets/LeetCode/BuySellStock/BuySellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int* left = &prices[0];
        int* right = &prices[1];
        int maxProfit = 0;

        if (prices.empty()) {
            return 0;
        }

        while (right!= prices.data() + prices.size()) {
            if (*right <= *left) {
                left = right;
            }
            else {
                maxProfit = max(maxProfit, *right - *left);
            }
            ++right;
        }

        return maxProfit;
    }
};

File: Problem_Sets/LeetCode/BuySellStock/BuySellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left, right = 0, 1
        maxProfit = 0
        while right < len(prices):
            if prices[right] <= prices[left]:
                left = right
            else:
                maxProfit = max(maxProfit, prices[right] - prices[left])
            right += 1
        return maxProfit